### DISCUSSION TITLE
Are there any mod incompatibilities? [Fabric 1.20.1]

### DISCUSSION THREAD

**Discussion by stahlfaust** [axieum/minecord#151] [Q&A] [answered]
Every function seems to be working on my server _except_ the in-game chat > Discord. I don't see anything in the server logs that stands out and I'm not sure why this feature isn't working for me.

**Comment by axieum** [comment]
Hi, do you have a list of mods on your server?

This mod uses the `fabric-message-api-v1` to detect in-game messages, so there shouldn't be any incompatibilities.
https://github.com/FabricMC/fabric/blob/e541ce876ccb928afd1c777fbbe9b53851df431f/fabric-message-api-v1/src/main/java/net/fabricmc/fabric/api/message/v1/ServerMessageEvents.java#L92-L105

Do you have the correct Discord channel id in your `config/minecord/chat.json5` file, and the bot has permission to send messages?

**Comment by stahlfaust** [comment]
Here's a mod list.
![image](https://github.com/axieum/minecord/assets/135766383/3a63c35b-4d29-4912-9286-4bb24f042f6d)

I do have the chat config set up properly. As you can see the bot is working for server events but when a player sends a chat message nothing happens. If I send a message through Discord it does show up in the Minecraft chat though.
![image](https://github.com/axieum/minecord/assets/135766383/d7068c03-ced6-4a02-a9df-b8afc48a249c)

Also if the bot has administrative permissions it should have message permissions I believe.

**Reply by axieum** [reply, marked as answer]
Hi, 👋 

I had a look into that "beautifiedchatserver" mod. I believe that is the culprit, it is injecting into where the chat message is received, sending its own new message and then cancelling the original message. Fabric is therefore never aware of the chat message and other mods can't pick it up.

https://github.com/Serilum/Collective/blob/1.21.0/Fabric%2Fsrc%2Fmain%2Fjava%2Fcom%2Fnatamus%2Fcollective%2Ffabric%2Fmixin%2FServerGamePacketListenerImplMixin.java#L32

I would recommend trying out Styled Chat for a much better experience.

**Reply by stahlfaust** [reply]
Thank you for looking into this. I will replace Beautified Chat with Styled Chat and post the results.

**Reply by stahlfaust** [reply]
That mod was indeed the issue! I guess you have an incompatibility list now! 😂

**Reply by axieum** [reply]
I'd say it's an incompatibility on their end, Minecord is not at fault.

**Reply by stahlfaust** [reply]
Well thank you for the help. I did have one question about the bot's configuration file. Is there a way to split chat and server events in to different Discord guild channels? I couldn't really find any wiki for the mod so wasn't sure if it was possible.
